src/main/java/com/griddynamics/cd/nrp/internal/uploading/impl/ArtifactUpdateApiClientImpl.java

```diff
@@ -119,14 +151,14 @@ public void onComplete(Future<RestResponse> future) throws InterruptedException
 
     /**
      * Creates jersey HTTP client
-     * @param login Username on the remote server
+     *
+     * @param login    Username on the remote server
      * @param password User's password
      * @return HTTP client
      */
     private Client getClient(String login, String password) {
         ClientConfig config = new DefaultClientConfig();
         Client client = Client.create(config);
-        client.setExecutorService(asyncRequestsExecutorService);
```

Comment:
@gd-tmagrys, I am not sure but I think that all Clients will not use the same thread executor. When I tested code without this line separate executor was created for the each client. So that we can not limit threads count.